runtime/sema/check_invocation_expression.go

```diff
@@ -459,6 +459,11 @@ func (checker *Checker) checkInvocation(
 		}
 	}
 
+	invocationRange := ast.NewRangeFromPositioned(
+		checker.memoryGauge,
+		invocationExpression,
+	)
```

Comment:
The range is now always allocated, but only used if the function type has a `ArgumentExpressionsCheck` or `TypePrametersCheck`. Maybe make this a lazy/memoized, so it's only allocated if needed, once:


```suggestion
	// Compute the invocation range, once, if needed
	getInvocationRange := func() func() ast.Range {
		var invcocationRange ast.Range
		return func() ast.Range {
			if invocationRange == ast.EmptyRange {
				invocationRange = ast.NewRangeFromPositioned(
					checker.memoryGauge,
					invocationExpression,
				)
			}
			return invocationRange
		}
	}()
```